DSCResources/MSFT_xWebsite/MSFT_xWebsite.psm1

```diff
@@ -261,59 +238,91 @@ function Set-TargetResource
+                }
+
+                # Add Default Pages for the newly created website
+                if ($PSBoundParameters.ContainsKey('DefaultPage') -and $DefaultPage -ne $null)
                 {
                     Update-DefaultPage -Name $Name -DefaultPage $DefaultPage
                 }
 
-                Write-Verbose -Message "successfully created website $Name"
+                Write-Verbose -Message "Successfully created website '$Name'."
 
-                #Start site if required
+                # Start website if required
                 if ($State -eq 'Started')
                 {
-                    #Wait 1 sec for bindings to take effect
-                    #I have found that starting the website results in an error if it happens to quickly
-                    Start-Sleep -Seconds 1
-                    Start-Website -Name $Name -ErrorAction Stop
-                }
+                    # Ensure that there are no other websites with binding information that will conflict with this site before starting
+                    if (-not (Confirm-UniqueBinding -BindingInfo $BindingInfo -ExcludeSite $Name))
+                    {
+                        # Return error and do not start Website
+                        $ErrorId = 'WebsiteBindingConflictOnStart'
+                        $ErrorCategory = [System.Management.Automation.ErrorCategory]::InvalidResult
+                        $ErrorMessage = $($LocalizedData.WebsiteBindingConflictOnStartError) -f ${Name}
+                        $Exception = New-Object -TypeName System.InvalidOperationException -ArgumentList $ErrorMessage
+                        $ErrorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord -ArgumentList $Exception, $ErrorId, $ErrorCategory, $null
+
+                        $PSCmdlet.ThrowTerminatingError($ErrorRecord)
+                    }
 
-                Write-Verbose -Message "successfully started website $Name"
+                    try
+                    {
+                        Start-Website -Name $Name -ErrorAction Stop
+
+                        Write-Verbose -Message "Successfully started website '$Name'."
+                    }
+                    catch
+                    {
+                        $ErrorId = 'WebsiteStateFailure'
+                        $ErrorCategory = [System.Management.Automation.ErrorCategory]::InvalidOperation
+                        $ErrorMessage = $($LocalizedData.WebsiteStateFailureError) -f ${Name}
+                        $ErrorMessage += $_.Exception.Message
+                        $Exception = New-Object -TypeName System.InvalidOperationException -ArgumentList $ErrorMessage
+                        $ErrorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord -ArgumentList $Exception, $ErrorId, $ErrorCategory, $null
+
+                        $PSCmdlet.ThrowTerminatingError($ErrorRecord)
+                    }
+                }
             }
             catch
```

Comment:
As a thought, what are we hoping to ctach here? We have try/catches within try/catches which slows down performance and can just obscure errors. Do we think we really need this?